Use rich logging for Pelican server

MinchinWeb opened the pull request:
Further to #2897, this enables *rich* logging for `pelican.server.

![image](https://user-images.githubusercontent.com/1548809/135697236-b6d496d7-6d9b-47a0-9977-6a0e075cffa0.png)

Surprisingly (at least to me), all that was needed was to import `console` from `pelican.log`.

Note this is working for `python -m pelican.server`, but doesn't seem to work on `pelican -l`. The latter seems to be drawing its logging from elsewhere.

P.S. If this could be marked "hacktoberfest-accepted" (so it counts towards my Hacktoberfest Pull Requests), that would be appreciated!

avaris commented:
* Linter is not happy (unused import)
* `pelican -l` bits are [here](https://github.com/getpelican/pelican/blob/f862d64b7a492f2af24e1228790a1d75371a1112/pelican/__init__.py#L478)
* This is a very limited support for `server`. All the messages the server outputs are rather "poor". It can be "enriched" by overriding [`log_message`](https://docs.python.org/3/library/http.server.html#http.server.BaseHTTPRequestHandler.log_message) method in the [subclass](https://github.com/getpelican/pelican/blob/f714f27c787a61e488ce922777ae322de7746cf6/pelican/server.py#L42) and using logging to display the output.

MinchinWeb commented:
Thanks for the tips @avaris ; I knew initially it was somewhat limited. I've fixed the three items you pointed to!

![image](https://user-images.githubusercontent.com/1548809/135718864-45a721b7-1f81-4f95-ab7d-1859d7d780c8.png)

Previously the logging provided both the local address and the time. Time is still there (although without the date) but I've dropped the local address as it is provided on starting and should never change; plus is keeps the log messages more on point.

Are we good to merge?

avaris commented on pelican/server.py:
```diff
@@ -99,6 +100,9 @@ def guess_type(self, path):
         return mimetype
 
+    def log_message(self, format, *args):
+        logger.info("%s" % format % args)
```
Extra formatting is unnecessary. This should be enough:
```python
logger.info(format, *args)
```

MinchinWeb replied:
Fixed! The previous code is what is used upstream.

MinchinWeb commented:
I've also set the minimum logging level to INFO, so that the server requests are actually shown.

avaris reviewed (approved):
Thanks

justinmayer commented:
@MinchinWeb: I don't have time today to check this in detail, but can you confirm that the output is not hard-wrapped? c.f., https://rich.readthedocs.io/en/latest/console.html#soft-wrapping

I ask because I think I noticed the other day that the new Rich-powered logging is now hard-wrapped, and if it is indeed now hard-wrapped, I think we should change that behavior as well.

justinmayer commented:
Disregard my last comment. The culprit was an outdated `tasks.py` file that did not have PTY set correctly.

Many thanks to @MinchinWeb for continuing to make Pelican's output richer and to @avaris for the review. 👏

justinmayer commented:
@MinchinWeb: One more thing... Could you squash the first four commits into one commit, leaving the "set minimum logging level to INFO" commit as a separate commit? That way, there should be two discrete commits in this PR once your squash is completed. (Normally I would do this manually outside the PR, but I want to ensure you get your Hacktoberfest credit.) Thank you!

justinmayer commented:
@MinchinWeb: Any chance you could squash four of the five commits here, as noted above? I'd like to publish a follow-up release today, and it'd be nice to include this while ensuring you get Hacktoberfest credit.

justinmayer commented:
Thanks again, @MinchinWeb!
